Build failure: hledger-utils

### Steps To Reproduce

Steps to reproduce the behavior:
1. `cd nixpkgs && git checkout 592047fc9e4f7b74a4dc85d1b9f5243dfe4899e3`  # nixpkgs-unstable @ 2023-11-15
2. `nix-build -A hledger-utils`

### Build log

```
Executing pythonImportsCheckPhase                                                                                                                                                                                                             
unittestCheckPhase                                                                                                                                                                                                                            
Executing unittestCheckPhase                                                                                                                                                                                                                  
Matplotlib created a temporary cache directory at /build/matplotlib-ki68f7uw because the default path (/homeless-shelter/.config/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variabl
e to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.                                                                                                                          
..F...........<frozen runpy>:128: RuntimeWarning: 'hledger_utils.commands.plot' found in sys.modules after import of package 'hledger_utils.commands', but prior to execution of 'hledger_utils.commands.plot'; this may result in unpredictab
le behaviour                                                                                                                                                                                                                                  
...                                                                                                                                                                                                                                           
======================================================================                                                                                                                                                                        
FAIL: test_hledger_edit_forecasted_not_edited (tests.test_hledger_edit.HledgerEditTest.test_hledger_edit_forecasted_not_edited)                                                                                                               
----------------------------------------------------------------------                                                                                                                                                                        
Traceback (most recent call last):                         
  File "/build/source/tests/test_hledger_edit.py", line 392, in test_hledger_edit_forecasted_not_edited                                                                                                                                       
    self.assert_hledger_edit(                              
  File "/build/source/tests/test_hledger_edit.py", line 93, in assert_hledger_edit                                                                                                                                                            
    self.assertEqual(                                      
AssertionError: '\n20[65 chars]    SNAACKSSSSS!!!\n    Cash                -2[31 chars]!!\n' != '\n20[65 chars]    Snacks\n    Cash                -20 €\n   [15 chars]ks\n'
                                                           
  2022-12-12  Food                                         
      Cash        -30 €                                    
      Cost:Food                                            
                                                           
- ~ monthly    SNAACKSSSSS!!!                              
+ ~ monthly    Snacks                                      
      Cash                -20 €                            
-     Cost:Food:SNAACKSSSSS!!!                             
+     Cost:Food:Snacks                                     
 : File 1 ('/build/hledger-edit-test-knzjzf7o.hledger') has wrong content after hledger-edit                                                                                                                                                  

----------------------------------------------------------------------                                                 
Ran 17 tests in 2.280s                                     

FAILED (failures=1)                                        
/nix/store/bbxdw4rgwwl3gnajri82yidr1nlsfskf-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context                                                                                                    
```

### Additional context

It seems to have been broken since august: https://hydra.nixos.org/build/240874754

### Notify maintainers

@NobbZ 

### Metadata

NixOS x86_64-linux.

--- a/pkgs/development/python-modules/hledger-utils/default.nix
+++ b/pkgs/development/python-modules/hledger-utils/default.nix
@@ -17,7 +17,7 @@
 
 buildPythonPackage rec {
   pname = "hledger-utils";
-  version = "1.13.2";
+  version = "1.14.0";
 
   format = "pyproject";
 
@@ -25,7 +25,7 @@ buildPythonPackage rec {
     owner = "nobodyinperson";
     repo = "hledger-utils";
     rev = "refs/tags/v${version}";
-    hash = "sha256-E++gHvXspPodLW2hEESKjg+R5aV39UJpDP8JYQX9iBw=";
+    hash = "sha256-Qu4nUcAGTACmLhwc7fkLxITOyFnUHv85qMhtViFumVs=";
   };
 
   SETUPTOOLS_SCM_PRETEND_VERSION = version;

